### PR TITLE
Feature/image pre process

### PR DESCRIPTION
**Pull request by GusAlejandro**
Pivotal task https://www.pivotaltracker.com/story/show/137327295

### CODE REVIEW

File: ImP/preProcessing/manipulate.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# IGNORE FILE
```
Comment (andrewtran1995) [inline ImP/preProcessing/manipulate.py]: I'd rename this file to something to reflect the fact that it's for prototyping.

File: ImP/preProcessing/manipulate.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+directory = testSettings.get_test_files_path()	
+pathForTestImage = os.path.join(directory,'brighter.jpg')
+
+processor = PreProcessor(pathForTestImage)
+temp = processor.pilImage.copy()
+#processor.pilImage.save('taco.jpg')
+processor.brighten_image(1.5)
+#temp.save(processor.path)
+
+"""
```
Comment (andrewtran1995) [inline ImP/preProcessing/manipulate.py]: I think these are usually included at the top of the file.

File: ImP/preProcessing/preProcessor.py [hunk 3]
```diff
@@ -1,24 +1,58 @@
...
+    def __init__(self, pathToNewImage):
+        self.image = cv2.imread(pathToNewImage) # for OpenCV
+        self.scikitImage = data.imread(pathToNewImage) # for scikit
+        self.pilImage = Image.open(pathToNewImage) # for PIL
+        self.path = pathToNewImage # used to save modified images to the same location, overriding them essentially
+
+    def is_similar(self, pathToOtherImage):
+        existingImage = cv2.imread(pathToOtherImage)
+        difference = cv2.subtract(self.image, existingImage)
+        self.pilImage.close()
```
Comment (andrewtran1995) [inline ImP/preProcessing/preProcessor.py]: Seems weird that the pilImage.close() step is done in this function, and not in some "tear down" function. Perhaps there should be a function just for that purpose?

Or, you could preface that instances of this class should always be used with the *with* statement to ensure proper resource cleanup.

File: ImP/preProcessing/preProcessor.py [hunk 4]
```diff
@@ -1,24 +1,58 @@
...
+        return np.mean(difference) <= self.image_similarity_threshold
+
+    def darken_image(self, darkFactor):
+        """
+        By setting the darkfactor to anything less than 1.0 we can darken the image.
+        """
+        self.pilImage.point(lambda x: x*darkFactor).save(self.path)
```
Comment (andrewtran1995) [inline ImP/preProcessing/preProcessor.py:32]: Yaaaaay lambdas <3

File: ImP/preProcessing/testImagePreProcess.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import os 
+from .preProcessor import PreProcessor 
+from .testSettings import testSettings
+
+
+class ImagePreProcessTestCase(unittest.TestCase):
+    def setUp(self):
+        """
+        pathForTestImage is for image similarity
+
```
Comment (andrewtran1995) [inline ImP/preProcessing/testImagePreProcess.py]: Could probably remove these newlines to make the file more compact.

File: ImP/preProcessing/testImagePreProcess.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        self.processor is for all of these
+
+        -------
+
+        pathForDark is for darken_image testcase,
+
+        self.processor2 can be used for both, so same image essentially
+
+        """
+        self.directory = testSettings.get_test_files_path()
```
Comment (andrewtran1995) [inline ImP/preProcessing/testImagePreProcess.py]: I like how all of these are handled in the setUp and tearDown.

**Review by andrewtran1995 — CHANGES_REQUESTED** [review]
Thanks for linking to the PIvotal task. Seems pretty clean, though I did note that the images in the README are broken for ImP/preProcessing/. If you want to make sure those images aren't broken, that'd be cool too. Otherwise, good stuff. :+1:

File: ImP/preProcessing/manipulate.py (hunk 1 above)
Comment (GusAlejandro) [reply ImP/preProcessing/manipulate.py]: LOL. Meant to delete this file.

File: ImP/preProcessing/preProcessor.py (hunk 3 above)
Comment (GusAlejandro) [reply ImP/preProcessing/preProcessor.py]: already closing the pilImage in teardown so this is no longer necessary

**Comment by GusAlejandro** [comment]
Made the changes, ready to be looked at once again. 
@andrewtran1995

**Comment by andrewtran1995** [comment]
Seems good, Ima merge it.

**Pull request merged**